pos_round_cash_payment_line/static/src/js/pos_round_cash_payment_line.js

```diff
@@ -26,92 +28,50 @@ odoo.define(
-            var change = order_prototype.get_change.call(this, paymentline);
-            if (this.pos.config.cash_rounding_activated){
-                return this.round_5c(change);
-            } else {
-                return change;
-            }
-        },
-
-        add_paymentline: function(cashregister) {
-            this.assert_editable();
-            var newPaymentline = new models.Paymentline(
-                {},
-                {
-                    order: this,
-                    cashregister:cashregister, pos: this.pos
-                }
-            );
-            if(cashregister.journal.type !== 'cash') {
-                newPaymentline.set_amount(this.get_due(newPaymentline))
-
-            } else if ( this.pos.config.iface_precompute_cash )  {
-                const due = this.get_due(newPaymentline);
-                newPaymentline.set_amount( this.round_5c(due) );
-
-            } else {
-                newPaymentline.set_amount( 0 );
-            }
-            this.paymentlines.add(newPaymentline);
-            this.select_paymentline(newPaymentline);
-        },
-
-        is_paid: function(){
-            if (this.pos.config.cash_rounding_activated
-                  && this.is_paid_with_cash()) {
-                return this.get_due() < 0.025;
-            } else {
-                return this.get_due() === 0;
+                );
+	            
+	            newPaymentline.set_amount(remainder);
+	            this.paymentlines.add(newPaymentline);
+	            this.select_paymentline(newPaymentline);
             }
         },
-
-        add_round_line: function (remainder) {
-            var remainder_product = this.pos.db.get_product_by_id(
-                this.pos.config.round_remainder_product_id[0]
-            );
-            var lines = this.get_orderlines();
-
-            for (var i = 0; i < lines.length; i++) {
-                if (lines[i].get_product() === remainder_product) {
-                    lines[i].set_unit_price(remainder);
-                    return;
-                }
-            }
-            this.add_product(remainder_product, {quantity: 1, price: remainder});
-        }
     });
-
+    
     screens.PaymentScreenWidget.include({
-
-        finalize_validation: function () {
-            if (this.pos.config.cash_rounding_activated) {
-                var order = this.pos.get_order();
-                var due = order.get_due();
-
-                if ( order.round_5c_remainder(due) ) {
-                        order.add_round_line(order.round_5c_remainder(-due))
-                    }
-                }
-            return this._super()
-        }
+    	click_paymentmethods: function(id) {
+    		var cashregister = null;
+    		
+    		for ( var i = 0; i < this.pos.cashregisters.length; i++ ) {
+    			if ( this.pos.cashregisters[i].journal_id[0] === id ){
+    				cashregister = this.pos.cashregisters[i];
+    				break;
+    			}
+    		}
```

Comment:
What happens if no cashregister is defined in the POS?

Reply:
c'est défini lors du click sur le bouton de la méthode de paiement donc à priori ce n'est jamais vide.